### ISSUE TITLE
Rename .ctags to make it easier to see

### ISSUE THREAD

**Issue by bitterjug** [kbsymanz/ctags-elm#3]
.ctags is a hidden file. There's no reason for this to be a dot file in the repo, users can copy the contents to their .ctags file or other destination, or link to it with a different name. I'm using fresh. But it dosn't even show up in nerdtree until I turn on hidden files which is odd. I'm proposing to rename the file to `elm.ctags` and add a symlink to `.ctags` for backward compatibility. With a note in the readme to deprecate use of the hidden file. What do you think?

**Comment by bitterjug** [comment]
Actually, I think we can do even better.  I just upgraded to [universal ctags](http://ctags.io/) and I have the following in `~/.ctags.d/preload/elm.conf` and now I can get nested functions:

```
--langdef=Elm
--langmap=Elm:.elm
--regex-Elm=/^([[:lower:]][[:alnum:]_]+)[[:blank:]]*:[^:].*->.*/\1/f,function,Functions/{scope=set}
--regex-Elm=/^ +([[:lower:]][[:alnum:]_]+)[[:blank:]]*:[^:].*->.*/\1/f,function,Functions/{scope=ref}
--regex-Elm=/^([[:lower:]][[:alnum:]_]+)[[:blank:]]*:[^:][^-]+$/\1/c,constant,Constant Function/{scope=clear}
--regex-Elm=/^ +([[:lower:]][[:alnum:]_]+)[[:blank:]]*:[^:][^-]+$/\1/c,constant,Constant Function/{scope=ref}
--regex-Elm=/^port +([[:lower:]][[:alnum:]_]+)[[:blank:]]*:[^:]/\1/p,port,Port/{scope=clear}
--regex-Elm=/^type +([[:upper:]][[:alnum:]_]+)/\1/t,typedef,Type Definition/{scope=clear}
--regex-Elm=/^type[[:blank:]]+alias[[:blank:]]+([[:upper:]][[:alnum:]_]+)/\1/a,alias,Type Alias/{scope=clear}
```

And they work great with tagbar, which I guess is why we're doing all this in the first place:

```
        let g:tagbar_type_elm = {
            \ 'ctagstype' : 'elm',
            \ 'kinds'     : [
                \ 't:types:1:0',
                \ 'a:type aliases:1:0',
                \ 'c:constants:1:0',
                \ 'f:functions:0:0',
                \ 'p:ports:0:0'
                \ ],
            \ 'sro'      : ':',
            \ 'kind2scope' : {
                \ 'f' : 'function'
            \ }
        \ }
```

Sadly this breaks exuberant ctags which doesn't recognise the brace that introduces long options (`{scope=set}`, etc.). So it makes sense to keep these separate. In my fork Im going to create `ctags.d/preload/elm.conf` for this config.

**Comment by kbsymanz** [comment]
I agree with keeping the projects separate if the changes break Exuberant Ctags compatible `.ctags` files. I haven't used Universal Ctags or Tabbar yet, though I will likely look into them soon. If the changes were not incompatible or were reasonable to manage with some clear instructions, I would consider incorporating the Universal and Tagbar code into this project _if that is your preference_, or you could cover that material in your own project and I could update my README to recommend your project for those interested in that flavor of ctags, etc.

Regarding renaming `.ctags` to `ctags`, my tendency is to not make that change for the sake of those who are just getting started and don't realize that they have to rename the file to make it work. I'd be open to changing my mind though if there were a number of other people who thought that the change was good, but I'm not seeing it yet.

I appreciate all of your thought and work along these lines. Thanks,

**Comment by bitterjug** [comment]
Hi Kurt

Thanks for this feedback. I'm kinda keen to have tags for nested functions, which is supported by vim-tagbar. When I suggested keeping the two versions of the ctags options separate I was thinking of separate files or folders in this repo. But I have taken your suggestion of managing a separate one because it's very convenient for me to structure it as a vim plugin which can refer to its own ctags options without intalling them in .ctags or similar. (tagbar is the only way I use ctags). So I have set up another project as an experiment with a view to merging the content into this or one of the other relevant projects later when Universal Ctags has grown up a bit. I've included a link to this project from the readme.

Thanks again for all your work and encouragement. 

Mark
